### PR TITLE
Fix virtualization ForeignKey field names and related names

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #3069
# What's Changed

- Rename `Cluster.type` to `cluster_type`
- Rename `Cluster.group` to `cluster_group`
- Rename `VirtualMachine.local_context_data` to `local_config_context_data`
  - Due to shared base class, also changes `Device.local_context_data`
- Rename `VirtualMachine.local_context_schema` to `local_config_context_schema`
  - Due to shared base class, also changes `Device.local_context_schema`
- Rename `VirtualMachine.local_context_data_owner_content_type` to `local_config_context_data_owner_content_type`
  - Due to shared base class, also changes `Device.local_context_data_owner_content_type`
- Rename `VirtualMachine.local_context_data_owner_object_id` to `local_config_context_data_owner_object_id`
  - Due to shared base class, also changes `Device.local_context_data_owner_object_id`
- Rename `VirtualMachine.local_context_data_owner` to `local_config_context_data_owner`
  - Due to shared base class, also changes `Device.local_context_data_owner`
- Change `VirtualMachine.local_context_schema` related_name from `virtualmachine_set` to `virtualization_virtualmachine_related`
  - Due to shared base class, also changes `Device.local_context_schema` 
- Change `VirtualMachine.local_context_data_owner_content_type` related_name from `virtualmachine_set` to `virtualization_virtualmachine_related`
  - Due to shared base class, also changes `Device.local_context_data_owner_content_type` 
- Rename `VMInterface.parent_interface` to `parent`
  - Due to shared base class, also changes `Interface.parent_interface`
- Change `VMInterface.parent_interface` related_name from `child_interfaces` to `children`
  - Due to shared base class, also changes `Interface.parent`
- Remove `parent` property from `Interface`
  - This shadowed the `Interface.device` field
- Remove `parent` property from `VMInterface`
  - This shadowed the `VMInterface.virtual_machine` field
  - Change `VirtualMachine.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/virtualization/tests/test_filters.py [hunk 1]
```diff
@@ -39,8 +39,8 @@ def setUpTestData(cls):
         )
 
         cls.clusters = [
```
Comment (HanlinMiao) [inline nautobot/virtualization/tests/test_filters.py:41]: We should probably change the name of `test_group` and `test_type` in `ClusterTestCase` in this file to `test_cluster_group` and `test_cluster_type`

File: nautobot/virtualization/filters.py [hunk 2]
```diff
@@ -91,21 +91,23 @@ class ClusterFilterSet(NautobotFilterSet, LocatableModelFilterSetMixin, TenancyM
...
+        field_name="cluster_type",
         queryset=ClusterType.objects.all(),
         label="Cluster type (ID)",
     )
     type = django_filters.ModelMultipleChoiceFilter(
-        field_name="type__slug",
+        field_name="cluster_type__slug",
         queryset=ClusterType.objects.all(),
         to_field_name="slug",
         label="Cluster type (slug)",
```
[the remark this answers is not in the document]
Comment (HanlinMiao) [reply nautobot/virtualization/filters.py:113]: That would conflict with the filter fields below I think, but Im not sure what is the difference between these filters?
We have two `Cluster type (slug)/(ID)` filters with different `field_name` attributes?

[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/virtualization/filters.py:113]: I was going to rename the filters as part of this change but I guess I missed this one. Thanks @glennmatthews

[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/virtualization/filters.py:113]: @HanlinMiao those filters are in `VirtualMachineFilterSet` and these are in `ClusterFilterSet` so no conflict there

File: nautobot/dcim/models/device_components.py [hunk 3]
```diff
@@ -792,10 +788,6 @@ def is_lag(self):
     def count_ipaddresses(self):
         return self.ip_addresses.count()
 
-    @property
-    def parent(self):
-        return self.device
-
```
[the remark this answers is not in the document]
Comment (gsnider2195) [reply nautobot/dcim/models/device_components.py]: Unfortunately I had to put this back. This is coming from the `CableTermination` base class so we would have to update `ConsolePort`, `ConsoleServerPort`, `PowerPort`, `PowerOutlet`, `FrontPort` and `RearPort` before we can remove/rename this on `Interface` and `VMInterface`. I still think it should be renamed but I don't think all of those changes are in scope for this pr.

File: nautobot/extras/models/models.py [hunk 4]
```diff
@@ -208,23 +212,27 @@ def get_config_context(self):
...
+        if self.local_config_context_data and not isinstance(self.local_config_context_data, dict):
+            raise ValidationError(
+                {"local_config_context_data": 'JSON data must be in object form. Example: {"foo": 123}'}
+            )
 
-        if self.local_context_schema and not self.local_context_data:
-            raise ValidationError({"local_context_schema": "Local context data must exist for a schema to be applied."})
+        if self.local_config_context_schema and not self.local_config_context_data:
+            raise ValidationError(
+                {"local_config_context_schema": "Local context data must exist for a schema to be applied."}
```
Comment (gsnider2195) [inline nautobot/extras/models/models.py:231]:
```suggestion
                {"local_config_context_schema": "Local config context data must exist for a schema to be applied."}
```

File: nautobot/virtualization/templates/virtualization/vminterface.html [hunk 5]
```diff
@@ -28,7 +28,7 @@
                 <tr>
                      <td>Parent Interface</td>
                      <td>
-                         {% if object.parent %}
+                         {% if object.parent_interface %}
```
Comment (gsnider2195) [inline nautobot/virtualization/templates/virtualization/vminterface.html:31]: This fixes a bug in the vminterface detail view

File: nautobot/virtualization/tests/test_filters.py [hunk 6]
```diff
@@ -722,11 +722,11 @@ def test_child(self):
             ),
         )
         VMInterface.objects.bulk_create(child_interfaces)
-        with self.subTest("Child Interfaces"):
+        with self.subTest("Children"):
```
Comment (gsnider2195) [inline nautobot/virtualization/tests/test_filters.py:725]:
```suggestion
        with self.subTest("Child Interfaces"):
```

File: nautobot/virtualization/tests/test_filters.py [hunk 7]
```diff
@@ -722,11 +722,11 @@ def test_child(self):
...
             params = {"child_interfaces": [child_interfaces[0].pk, child_interfaces[1].name]}
             self.assertEqual(self.filterset(params, self.queryset).qs.count(), 1)
 
-        with self.subTest("Has child Interfaces"):
+        with self.subTest("Has children"):
```
Comment (gsnider2195) [inline nautobot/virtualization/tests/test_filters.py:729]:
```suggestion
        with self.subTest("Has child Interfaces"):
```

**Review by glennmatthews — APPROVED** [review]
Assuming tests pass, this looks good to go. 👍

**Pull request merged**